src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ClusterMonitorSupervisor.scala

```diff
@@ -134,20 +149,55 @@ class ClusterMonitorSupervisor(monitorConfig: MonitorConfig, dataprocConfig: Dat
     }
   }
 
-  def autoFreezeClusters: Future[Unit] = {
+  def autoFreezeClusters(): Future[Unit] = {
     val clusterList = dbRef.inTransaction {
       _.clusterQuery.getClustersReadyToAutoFreeze()
     }
     clusterList map { cl =>
       cl.foreach { c =>
         logger.info(s"Auto freezing cluster ${c.clusterName} in project ${c.googleProject}")
-        leoService.internalStopCluster(c).failed.foreach { e =>
+        leonardoService.internalStopCluster(c).failed.foreach { e =>
           logger.warn(s"Error occurred auto freezing cluster ${c.projectNameString}", e)
         }
       }
     }
   }
 
+  private def createClusterMonitors(implicit executionContext: ExecutionContext): Unit = {
+    dbRef
+      .inTransaction { _.clusterQuery.listMonitoredFullCluster() }
+      .onComplete {
+        case Success(clusters) =>
+          val clustersNotAlreadyBeingMonitored = clusters.toSet -- monitoredClusters.value
+
+          clustersNotAlreadyBeingMonitored foreach {
+            case c if c.status == ClusterStatus.Deleting => {
+              logger.info("deleting cluster")
+              self ! ClusterDeleted(c)
+            }
+            case c if c.status == ClusterStatus.Stopping => {
+              logger.info("stopping cluster")
+              self ! ClusterStopped(c)
+            }
+            case c if c.status == ClusterStatus.Starting => self ! ClusterStarted(c)
+
+            case c if c.status == ClusterStatus.Updating => self ! ClusterUpdated(c)
+
+            case c => {
+              logger.info("stop after creation::"  + c.stopAfterCreation)
+              self ! ClusterCreated(c, c.stopAfterCreation)
+            }
+          }
+          monitoredClusters.mutate(_ ++ clustersNotAlreadyBeingMonitored)
```

Comment:
Changing an actor's state from within a Future is usually not good practice. It would be better to pipe a message back to the actor with the clusters that need to be added to the monitored list.

Reply:
Or maybe the `ClusterStarted`, `ClusterStopped`, etc handlers take care of updating the monitored list themselves.